packages/@aws-cdk/cloudformation-diff/lib/diff/template-and-changeset-diff-merger.ts

```diff
@@ -0,0 +1,211 @@
+// The SDK is only used to reference `DescribeChangeSetOutput`, so the SDK is added as a devDependency.
+// The SDK should not make network calls here
+import type { DescribeChangeSetOutput as DescribeChangeSet, ResourceChangeDetail as RCD } from '@aws-sdk/client-cloudformation';
+import { diffResource } from '.';
+import * as types from '../diff/types';
+
+export type DescribeChangeSetOutput = DescribeChangeSet;
+export type ChangeSetResourceChangeDetail = RCD;
+
+/**
+ * The purpose of this class is to include differences from the ChangeSet to differences in the TemplateDiff.
+ */
+export class TemplateAndChangeSetDiffMerger {
+  // If we somehow cannot find the resourceType, then we'll mark it as UNKNOWN, so that can be seen in the diff.
+  static UNKNOWN_RESOURCE_TYPE = 'UNKNOWN';
+
+  /**
+   * @param parseOldOrNewValues These enum values come from DescribeChangeSetOutput, which indicate if the property resolves to that value after the change or before the change.
+   */
+  static convertResourceFromChangesetToResourceForDiff(
```

Comment:
Seems like this is only ever used by other class methods so `private static convertResourceFromChangesetToResourceForDiff`?

Reply:
I will add the public and private modifiers. However, some of these functions, while only used in this class for application code, are directly unit tested. So I will set those methods as public so that they can be unit tested